src/mon/MonClient.cc

```diff
@@ -463,6 +463,7 @@ int MonClient::authenticate(double timeout)
   if (active_con) {
     ldout(cct, 5) << __func__ << " success, global_id "
 		  << active_con->get_global_id() << dendl;
+    authenticated = true;
```

Comment:
could you help add `assert(!authenticate_err)` here also ? it took me a while to convince myself that a valid active_con implies `authenticate_err == 0`.